graphql_api/types/comparison/comparison.py

```diff
@@ -184,18 +184,30 @@ def resolve_flag_comparisons(
 @comparison_bindable.field("componentComparisons")
 @sync_to_async
 def resolve_component_comparisons(
-    comparison_report: ComparisonReport, info
+    comparison_report: ComparisonReport, info, filters=None
 ) -> List[ComponentComparison]:
     current_owner = info.context["request"].current_owner
     head_commit = comparison_report.commit_comparison.compare_commit
     components = components_service.commit_components(head_commit, current_owner)
+    list_components = comparison_report.commit_comparison.component_comparisons.all()
+
+    if filters and filters.get("components"):
+        components = [
+            component
+            for component in components
+            if component.name in filters["components"]
+        ]
+
+        list_components = list_components.filter(
+            component_id__in=[component.component_id for component in components]
+        )
 
     # store for child resolvers (needed to get the component name, for example)
     info.context["components"] = {
         component.component_id: component for component in components
     }
 
-    return list(comparison_report.commit_comparison.component_comparisons.all())
+    return list(list_components)
```

Comment:
Thoughts on keeping component filter consistent with flag filtering where in flag filtering if there are no results at the end of filtering we would return an error (UnknownFlags) instead of an empty array?

Reply:
we huddled on this, the need to create a deprecate field isn't worth it so we're moving forward with this